rust-libretro/src/environment.rs

```diff
@@ -1000,31 +1068,33 @@ pub fn get_led_interface(callback: retro_environment_t) -> Option<retro_led_inte
 /// See [`AudioVideoEnable`] for descriptions of the flags.
 #[proc::context(GenericContext)]
 #[proc::unstable(feature = "env-commands")]
-pub unsafe fn get_audio_video_enable(callback: retro_environment_t) -> AudioVideoEnable {
+pub unsafe fn get_audio_video_enable(
+    callback: retro_environment_t,
+) -> Result<AudioVideoEnable, EnvironmentCallError> {
     // int *
-    if let Some((info, _)) = get(callback, RETRO_ENVIRONMENT_GET_AUDIO_VIDEO_ENABLE) {
-        return AudioVideoEnable::from_bits_truncate(info);
-    }
+    let info = get(callback, RETRO_ENVIRONMENT_GET_AUDIO_VIDEO_ENABLE)?;
 
-    AudioVideoEnable::empty()
+    validate_bitflags!(AudioVideoEnable, u32, info)
 }
 
 /// Returns a MIDI interface that can be used for raw data I/O.
 #[proc::context(GenericContext)]
 #[proc::unstable(feature = "env-commands")]
-pub fn get_midi_interface(callback: retro_environment_t) -> Option<retro_midi_interface> {
+pub fn get_midi_interface(
+    callback: retro_environment_t,
+) -> Result<retro_midi_interface, EnvironmentCallError> {
     // struct retro_midi_interface **
-    get_unchecked(callback, RETRO_ENVIRONMENT_GET_MIDI_INTERFACE).map(|(v, _)| v)
+    get_unchecked(callback, RETRO_ENVIRONMENT_GET_MIDI_INTERFACE)
 }
 
 /// Boolean value that indicates whether or not the frontend is in fastforwarding mode.
 #[proc::context(GenericContext)]
 #[proc::unstable(feature = "env-commands")]
-pub unsafe fn get_fastforwarding(callback: retro_environment_t) -> bool {
```

Comment:
`RETRO_ENVIRONMENT_GET_FASTFORWARDING` returning `false` is not an error or an indicator of a lack of support. It just means that the player isn't holding the "fast-forward" button. So I'd return a `Result<bool, EnvironmentCallError>`.